symfony/plugins/orangehrmTimePlugin/Api/CopyProjectActivityAPI.php

```diff
@@ -0,0 +1,189 @@
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Time\Api;
+
+use OrangeHRM\Core\Api\V2\CrudEndpoint;
+use OrangeHRM\Core\Api\V2\Endpoint;
+use OrangeHRM\Core\Api\V2\EndpointResourceResult;
+use OrangeHRM\Core\Api\V2\EndpointResult;
+use OrangeHRM\Core\Api\V2\Model\ArrayModel;
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Entity\Project;
+use OrangeHRM\Time\Api\Model\CopyActivityModel;
+use OrangeHRM\Time\Dto\CopyActivityField;
+use OrangeHRM\Time\Traits\Service\ProjectServiceTrait;
+
+class CopyProjectActivityAPI extends Endpoint implements CrudEndpoint
+{
+    use ProjectServiceTrait;
+
+    public const PARAMETER_FROM_PROJECT_ID = 'fromProjectId';
+    public const PARAMETER_TO_PROJECT_ID = 'toProjectId';
+    public const PARAMETER_ACTIVITY_IDS = 'activityIds';
+
+    /**
+     * @inheritDoc
+     */
+    public function getAll(): EndpointResult
+    {
+        throw $this->getNotImplementedException();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetAll(): ParamRuleCollection
+    {
+        throw $this->getNotImplementedException();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function create(): EndpointResult
+    {
+        list($toProjectId) = $this->getUrlAttributes();
+        $ids = $this->getRequestParams()->getArray(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_ACTIVITY_IDS);
+        $this->getProjectService()->getProjectActivityDao()->saveCopyActivity($toProjectId, $ids);
+        return new EndpointResourceResult(ArrayModel::class, $ids);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForCreate(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            $this->getValidationDecorator()->requiredParamRule(
+                new ParamRule(
+                    self::PARAMETER_ACTIVITY_IDS,
+                    new Rule(Rules::ARRAY_TYPE),
+                    new Rule(
+                        Rules::EACH,
+                        [new Rules\Composite\AllOf(new Rule(Rules::POSITIVE))]
+                    )
+                ),
+            ),
+            ...$this->getCommonBodyValidationRules(),
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function delete(): EndpointResult
+    {
+        throw $this->getNotImplementedException();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForDelete(): ParamRuleCollection
+    {
+        throw $this->getNotImplementedException();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getOne(): EndpointResult
```

Comment:
please move to getAll with pagination. It seems your result is a collection, but returning instance of `EndpointResourceResult` here

Reply:
done